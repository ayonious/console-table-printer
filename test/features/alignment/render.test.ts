import { Table } from '../../../index';
import { getTableBody, getTableHeader } from '../../testUtils/getRawData';

describe('Alignment Tests: Rendering', () => {
  it('should render left-aligned content', () => {
    const p = new Table({
      shouldDisableColors: true,
      columns: [
        { name: 'text', alignment: 'left' },
        { name: 'value', alignment: 'left' },
      ],
    });

    p.addRow({ text: 'Left Text', value: 123 });

    const [renderedHeader, renderedBody] = [getTableHeader(p), getTableBody(p)];
    expect(renderedHeader).toEqual('│ text      │ value │');
    expect(renderedBody).toEqual(['│ Left Text │ 123   │']);
  });

  it('should render right-aligned content', () => {
    const p = new Table({
      shouldDisableColors: true,
      columns: [
        { name: 'text', alignment: 'right' },
        { name: 'value', alignment: 'right' },
      ],
    });

    p.addRow({ text: 'Right Text', value: 123 });

    const [renderedHeader, renderedBody] = [getTableHeader(p), getTableBody(p)];
    expect(renderedHeader).toEqual('│       text │ value │');
    expect(renderedBody).toEqual(['│ Right Text │   123 │']);
  });

  it('should render center-aligned content', () => {
    const p = new Table({
      shouldDisableColors: true,
      columns: [
        { name: 'text', alignment: 'center' },
        { name: 'value', alignment: 'center' },
      ],
    });

    p.addRow({ text: 'Center Text', value: 123 });

    const [renderedHeader, renderedBody] = [getTableHeader(p), getTableBody(p)];
    expect(renderedHeader).toEqual('│    text     │ value │');
    expect(renderedBody).toEqual(['│ Center Text │  123  │']);
  });

  it('should render mixed alignments', () => {
    const p = new Table({
      shouldDisableColors: true,
      columns: [
        { name: 'left', alignment: 'left' },
        { name: 'right', alignment: 'right' },
        { name: 'center', alignment: 'center' },
      ],
    });

    p.addRow({ left: 'Left', right: 'Right', center: 'Center' });

    const [renderedHeader, renderedBody] = [getTableHeader(p), getTableBody(p)];
    expect(renderedHeader).toEqual('│ left │ right │ center │');
    expect(renderedBody).toEqual(['│ Left │ Right │ Center │']);
  });
});
