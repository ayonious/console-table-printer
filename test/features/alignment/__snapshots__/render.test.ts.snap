// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alignment Tests: Rendering should render center-aligned content 1`] = `
"┌───────────────────────────┬───────────────────────┐
│ centered_text_column_name │ centered_value_column │
├───────────────────────────┼───────────────────────┤
│        Center Text        │          123          │
│        Middle Row         │          456          │
│        Last Center        │          789          │
└───────────────────────────┴───────────────────────┘"
`;

exports[`Alignment Tests: Rendering should render left-aligned content 1`] = `
"┌────────────────────────────────────────┬────────────────────────────────┐
│ very_long_text_column_name_for_testing │ another_long_value_column_name │
├────────────────────────────────────────┼────────────────────────────────┤
│ Left Text                              │ 123                            │
│ Another Left                           │ 456                            │
│ Third Row                              │ 789                            │
└────────────────────────────────────────┴────────────────────────────────┘"
`;

exports[`Alignment Tests: Rendering should render mixed alignments 1`] = `
"┌─────────────────────┬──────────────────────┬───────────────────────┐
│ left_aligned_column │ right_aligned_column │ center_aligned_column │
├─────────────────────┼──────────────────────┼───────────────────────┤
│ Left Text           │           Right Text │      Center Text      │
│ Left2 Long          │          Right2 Long │     Center2 Long      │
│ Left3 Longer        │        Right3 Longer │    Center3 Longer     │
└─────────────────────┴──────────────────────┴───────────────────────┘"
`;

exports[`Alignment Tests: Rendering should render right-aligned content 1`] = `
"┌────────────────────────────────┬─────────────────────────────┐
│ long_text_column_right_aligned │ numeric_value_right_aligned │
├────────────────────────────────┼─────────────────────────────┤
│                     Right Text │                         123 │
│                     More Right │                         456 │
│                     Last Right │                         789 │
└────────────────────────────────┴─────────────────────────────┘"
`;
